keps/sig-node/20190505-pod-level-resource-quota.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Pod Level Resource Quota
+authors:
+  - "@resouer"
+  - "@qinguoan"
+  - "@zhangxiaoyu-zidif"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+reviewers:
+  - "@DawnChen"
+  - "@yujuhong"
+approvers:
+  - "@DawnChen"
+  - "@yujuhong"
+editor: TBD
+creation-date: 2019-05-05
+last-updated: 2019-05-05
+status: provisional
+---
+
+# Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Implementation History](#implementation-history)
+- 
+
+## Summary
+The purpose of this enhancement is to redefine QoS class and Pod resource capacity's measurement methods.
+
+In the current QoS system, all containers resources of a Pod will be considered when calculate the Pod QoS class and Pod resource capacity. In the proposal, sidecar containers' reources will be ignored in above calculations.
+
+## Motivation
+
+Now the current QoS class considers all containers' resources, but as for sidecar containers, its measurement is not quite proper. Sidecar container usuallly cost not much resources and meanwhile its resources is hard to allocated. So user do not define its resource when describing a pod commonly which would result in that the Pod is NOT guaranteed and has lower priority. We need to modify QoS class measurement to make a pod still be guaranteed without setting sidecar containers' resources.
+
+
+### Goals
+
+* Ignore sidecar containers' resources when calculate Pod's QoS class
```

Comment:
One thing I'm not clear on is how we will know if a container is a sidecar container or not?

Reply:
So we need to label this container and mark it as sidecar, so we push another KEP about sidecar. There are many relationships.